### DISCUSSION TITLE
Avoid too many Blank Lines

### DISCUSSION THREAD

**Discussion by christianbraeunlich** [microsoft/alguidelines#153] [BC Best Practices]
Started from https://github.com/StefanMaron/BusinessCentral.LinterCop/discussions/174

---

I'd like to avoid too many blank lines.

Inspired by:
- https://www.python.org/dev/peps/pep-0008/#blank-lines
- https://www.flake8rules.com/rules/E303.html

---

## Bad code

```al
var
    IsVisible: Boolean;
    IsEditable: Boolean;


local procedure Init()
begin
    IsVisible := false;
    IsEditable := false;


    exit(true);
end;
```

## Good code

```al
var
    IsVisible: Boolean;
    IsEditable: Boolean;

local procedure Init()
begin
    IsVisible := false;
    IsEditable := false;

    exit(true);
end;
```

---

We could also argue if at some point there has to be a certain number of blank lines for example between object properties and layout section.

**Comment by waldo1001** [comment]
Yep, very valid!
You want to describe it?

**Comment by fvet** [comment]
Maybe a Tips / Links sections can be added, pointing to AL/AZ Dev Tools?

https://github.com/anzwdev/al-code-outline/issues/347#issuecomment-1107530741

**Reply by christianbraeunlich** [reply]
Nice! Will use this myself from now on. I would love to see it in there. Would you like to open a PR?

**Reply by waldo1001** [reply]
That would be awesome indeed :-)

**Reply by fvet** [reply]
> Nice! Will use this myself from now on. I would love to see it in there. Would you like to open a PR?

Done ;)
